src/utils/markets.tsx

```diff
@@ -330,6 +343,28 @@ export function _useUnfilteredTrades(limit = 10000) {
   //   .map(market.parseFillEvent.bind(market));
 }
 
+export function useBonfidaTrades() {
+  const { market } = useMarket();
+  async function getBonfidaTrades() {
+    if (!market) {
+      return null;
+    }
+    const response = await fetch(
```

Comment:
We should wrap this in a try-catch and log the error to the console if the fetch fails

Reply:
Biggest worry here is rate limiting from the Bonfida api. I think for errors we should return the last successful fetched set of trades rather than nullifying the cache. IMO the thing that makes the most sense there is to create a version of `useAsyncData` that has an escape hatch to stop updating the global cache in case of a non-success error code.

Reply:
added an optional parameter `setCacheToNull` to avoid nullifying the cache. Seemed like a natural way of doing it